lib/livebook/hubs.ex

```diff
@@ -186,18 +186,67 @@ defmodule Livebook.Hubs do
   ## Example
 
       iex> get_connected_hubs()
-      [%{pid: #PID<0.178.0>, hub: %Enterprise{}}, ...]
+      [%Enterprise{}, ...]
 
   """
-  @spec get_connected_hubs() :: connected_hubs()
+  @spec get_connected_hubs() :: list(Provider.t())
   def get_connected_hubs do
-    for hub <- get_hubs(), connected = get_connected_hub(hub), do: connected
+    for hub <- get_hubs(),
+        capability?(hub, :connect),
+        Provider.connected?(hub),
+        into: [],
+        do: hub
+  end
+
+  @doc """
+  Gets a list of connected hubs with given capabilities.
+
+  ## Example
+
+      iex> get_connected_hubs([:secrets])
+      [%Enterprise{}, ...]
+
+  """
+  @spec get_connected_hubs(Provider.capabilities()) :: list(Provider.t())
+  def get_connected_hubs(capabilities) do
+    for hub <- get_connected_hubs(),
+        capability?(hub, capabilities -- [:connect]),
+        into: [],
+        do: hub
+  end
+
+  @doc """
+  Gets a list of hub secrets.
+
+  It gets from all hubs with secret management.
+  """
+  @spec get_secrets() :: list(Secret.t())
+  def get_secrets do
+    for hub <- get_connected_hubs([:secrets]),
```

Comment:
I don't think it has to be connected: 

```suggestion
    for hub <- get_hubs([:secrets]),
```

:)